File: paper_artifacts/experiments/llr9/artifacts/sources/oss120b-fortran/tsvc_2_s1244__submitted.f90
subroutine tsvc_2_s1244_fp64(a, b, c, d, len_1d, workspace, workspace_bytes) bind(C, name="tsvc_2_s1244_fp64")
  use iso_c_binding
  implicit none
  integer(c_int64_t), value :: len_1d
  type(c_ptr), value :: workspace
  integer(c_int64_t), value :: workspace_bytes
  real(c_double), intent(inout) :: a(*)
  real(c_double), intent(in) :: b(*), c(*)
  real(c_double), intent(out) :: d(*)
  integer(c_int64_t) :: i
  real(c_double), allocatable :: a_old(:)

  if (len_1d <= 1) then
    return
  end if

  allocate(a_old(len_1d))
  !$omp parallel do private(i) schedule(static)
  do i = 1, len_1d
    a_old(i) = a(i)
  end do
  !$omp end parallel do

  !$omp parallel do private(i) schedule(static)
  do i = 1, len_1d-1
    a(i) = b(i) + c(i)*c(i) + b(i)*b(i) + c(i)
  end do
  !$omp end parallel do

  !$omp parallel do private(i) schedule(static)
  do i = 1, len_1d-1
    d(i) = a(i) + a_old(i+1)
  end do
  !$omp end parallel do

  deallocate(a_old)
end subroutine tsvc_2_s1244_fp64

File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s1244__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s1244_fp64(double *restrict a, double *restrict b,
                       double *restrict c, double *restrict d,
                       int64_t n, uint8_t *restrict workspace,
                       int64_t workspace_bytes) {
    if (n <= 1) return;
    int64_t need = 8 * n;
    if (workspace && workspace_bytes >= need) {
        double *restrict olda = (double *restrict)workspace;
        #pragma omp parallel for simd schedule(static)
        for (int64_t i = 0; i < n; i++) {
            olda[i] = a[i];
        }
        #pragma omp parallel for simd schedule(static)
        for (int64_t i = 0; i < n - 1; i++) {
            double bi = b[i];
            double ci = c[i];
            double ai = bi + ci * ci + bi * bi + ci;
            a[i] = ai;
            d[i] = ai + olda[i + 1];
        }
    } else {
        for (int64_t i = 0; i < n - 1; i++) {
            double bi = b[i];
            double ci = c[i];
            double ai = bi + ci * ci + bi * bi + ci;
            a[i] = ai;
            d[i] = ai + a[i + 1];
        }
    }
}
